### ISSUE TITLE
Globals no longer supported

### ISSUE THREAD

**Issue by kentcb** [aws/aws-extensions-for-dotnet-cli#45]
I've upgraded from `2.2.0` to `3.1.0` (a big update, I know) and it's broken globals. For example, I have this in my template:

```yaml
Globals:
  Function:
    Runtime: "dotnetcore2.1"

# later 
  PingFunction:
    Type: "AWS::Serverless::Function"
    Properties:
      FunctionName: !Sub ${Environment}_Ping
      Handler: "LambdaFunctions::LambdaFunctions::Ping"
      Role: !GetAtt LambdaRole.Arn
```

But attempting deployment fails with:

```
Unknown error executing command: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Amazon.Lambda.Tools.LambdaUtilities.DetermineTargetFrameworkFromLambdaRuntime(String lambdaRuntime) in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Lambda.Tools\LambdaUtilities.cs:line 31
   at Amazon.Lambda.Tools.TemplateProcessor.TemplateProcessorManager.PackageDotnetProjectAsync(IUpdateResourceField field, String location) in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Lambda.Tools\TemplateProcessor\TemplateProcessorManager.cs:line 203
   at Amazon.Lambda.Tools.TemplateProcessor.TemplateProcessorManager.ProcessUpdatableResourceAsync(String templateDirectory, IUpdateResourceField field) in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Lambda.Tools\TemplateProcessor\TemplateProcessorManager.cs:line 155
   at Amazon.Lambda.Tools.TemplateProcessor.TemplateProcessorManager.TransformTemplateAsync(String templateDirectory, String templateBody) in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Lambda.Tools\TemplateProcessor\TemplateProcessorManager.cs:line 87
   at Amazon.Lambda.Tools.Commands.DeployServerlessCommand.PerformActionAsync() in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Lambda.Tools\Commands\DeployServerlessCommand.cs:line 167
   at Amazon.Common.DotNetCli.Tools.Commands.BaseCommand`1.ExecuteAsync() in E:\JenkinsWorkspaces\aws-extensions-for-dotnet-cli\src\Amazon.Common.DotNetCli.Tools\Commands\BaseCommand.cs:line 44
```

Debugging through the code, I can see it is because it expects, for example, a `Runtime` value on the `IUpdatableResourceDataSource`, but that is `null` when using a global. I tried adding it manually to my function and it was then present, but obviously I don't want to be doing that.

**Comment by normj** [comment]
Yes that was a miss, taking a look for a fix.

**Comment by normj** [comment]
I have a fix checked in on the dev branch https://github.com/aws/aws-extensions-for-dotnet-cli/commit/338a5cbe4bfb167d706b9c1c9aa126d594f95261.

I was planning on doing a release in a couple days. Are you blocked or can you wait a couple days for the fix to be a part of the next release?

**Comment by kentcb** [comment]
@normj Thanks for fixing this promptly. I am able to continue working for now with a hacked version of the tooling. As long as the fix is deployed within a couple of days it shouldn't affect me too much.

**Comment by kentcb** [comment]
@normj Do you have any further info on when the fix will be released? I'm at the point where this is now blocking me.

**Comment by normj** [comment]
I just push the fix in version 3.1.1 of Amazon.Lambda.Tools.
